src/components/App/AppIFrame.js

```diff
@@ -67,16 +62,37 @@ class AppIFrame extends React.Component {
     const { hidden, src } = this.props
     return !src || hidden
   }
+  navigateIFrame = src => {
+    // Rather than load src=undefined, this component hides itself. That way, if the user later
+    // navigates back to the same src, we don't have to reload the iframe.
+    if (src) {
+      // Cache src to avoid cases where the iframe would load the same page as before
+      this.src = src
+      this.setProgressTimeout(this.startProgress(), 100)
+
+      // Detach the iframe from the DOM before setting the src to avoid adding history state
```

Comment:
What about using `this.iframe.contentWindow.location.replace(src)` instead of detaching the node?

From https://stackoverflow.com/a/821373/292500

Reply:
Tried this first, but unfortunately using `location.replace()` doesn't call the `onload()` handler for some reason. I have no idea why.